fix: schema for agent.image.pullSecrets

tredmon opened the pull request:
## Description

The `agent.image.pullSecrets` is referenced by the [deployment.yaml](https://github.com/telepresenceio/telepresence/blob/release/v2/charts/telepresence-oss/templates/deployment.yaml#L174) but is currently disallowed by the schema file. I need to use this configuration setting in my k8s environment. I currently use a workaround of `helm upgrade --skip-schema-validation` but it would be nice not to need to skip validation.

## Checklist

<!--
  Please review the requirements for each checkbox, and check them
  off (change "[ ]" to "[x]") as you verify that they are complete.
-->

 - [x] I made sure to update `./CHANGELOG.yml` (our release notes are generated from this file).
 - [x] I made sure to add any documentation changes required for my change.
 - [x] My change is adequately tested.
 - [x] I updated `CONTRIBUTING.md` with any special dev tricks I had to use to work on this code efficiently.
 - [x] Once my PR is ready to have integration tests ran, I posted the PR in #telepresence-oss channel on the
       [CNCF Slack](https://slack.cncf.io/) so that the "ok to test" label can be applied.

thallgren commented:
This looks good to me, but please ensure two things:

1. Run `make docs-files` so that your addition gets included in the docs.
2. You must [sign off](https://github.com/telepresenceio/telepresence/pull/4014/checks?check_run_id=57737646728) on your commit.

Commit:
fix: schema for agent.image.pullSecrets

Signed-off-by: Trevan Redmon <trevan.redmon@gmail.com>

tredmon commented:
`make docs-files` only modified the default config path to it's location on MacOS in 86 different files in the docs folder. That didn't seem like a worthwhile change to commit.
```
-      --config string     Path to the Telepresence configuration file (default "$HOME/.config/telepresence/config.yml")
+      --config string     Path to the Telepresence configuration file (default "$HOME/Library/Application Support/telepresence/config.yml")
```

I've now signed off on the commit, and manually added missing documentation to the chart's README.md.

Oh, looks like `make docs-files` only runs against the most recent commit (either that or I just missed it among all the default config path changes). Since I had a commit for my change followed by a merge commit it didn't originally run against my change. Now that I've rebased everything on `release/v2` and my commit was the latest, running `make docs-files` did produce some useful changes. I've added those to my commit as well now.

thallgren commented:
The build is currently failing due to an external problem: https://gitlab.alpinelinux.org/alpine/aports/-/issues/17775. I expect it go get fixed very soon. I'll rerun the build and merge once that happens.
